### DISCUSSION TITLE
Mojo Docs Frameworks

### DISCUSSION THREAD

**Discussion by Collegehive** [modular/modular#2078] [General]
I wanted to know which framework does mojo docs runs on.

And where is the code that runs that mojo docs site

means i want to make a site similar site of mojo docs

**Comment by Brian-M-J** [comment]
I would also like to know about this. Particularly:
- Will Mojo just reuse [Sphinx](https://www.sphinx-doc.org/en/master/)/[Docutils](https://docutils.sourceforge.io/) or make another documentation generator?
- What text format will it use? [reST](https://docutils.sourceforge.io/rst.html) or [MyST](https://myst-parser.readthedocs.io/en/latest/)?

**Comment by iamtimdavis** [comment]
@scottamain can provide more context here.

But the TLDR is: we built a custom stack for our docs site, and we won't be open sourcing it anytime soon (we just have so much going on, its too hard to maintain given we just OSS'd Mojo🔥 stdlib among other things). We appreciate your interest however - and you can contribute to improving our docs via https://github.com/modularml/mojo/tree/main/docs.

**Comment by scottamain** [comment]
I'd be happy to answer any specific questions, but here are the basics: The website application is built with Docusaurus (with a lot of customization); the Mojo standard library is generated with [`mojo doc`](https://docs.modular.com/mojo/cli/doc) and converted to markdown (from JSON); the Mojo Manual is mostly written in Jupyter notebooks ([source is here](https://github.com/modularml/mojo/tree/main/docs/manual)) and converted to markdown with `nbconvert`.

Currently, we have no plans to use Sphinx for Mojo. Mojo has a lot of unique syntax, so `mojo doc` is a custom process that generates structured content (JSON) from the API docstrings using the Mojo parser. This has many limitations compared to generating Python API docs with Sphinx, but improving this tooling hasn't been a priority for the team yet. I'd like to discuss the long-term options for Mojo API docs soon, and would love to hear ideas and suggestions.
